pkg/client/vault/client.go

```diff
@@ -205,3 +304,95 @@ func compressZCAP(zcap *zcapld.Capability) (string, error) {
 
 	return base64.URLEncoding.EncodeToString(compressed.Bytes()), nil
 }
+
+func lastElm(s, sep string) string {
+	all := strings.Split(s, sep)
+
+	return all[len(all)-1]
+}
+
+func buildEDVURI(h, vid, did string) string {
+	return fmt.Sprintf("%s/encrypted-data-vaults/%s/documents/%s", h, vid, did)
+}
+
+func prepareRemoteJWEKey(keyURL string, kmsImpl kms.KeyManager) ([]byte, string, error) {
+	kid := lastElm(keyURL, "/")
+
+	pubKeyBytes, err := kmsImpl.ExportPubKeyBytes(kid)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to retrieve main public key bytes from remote KMS: %w", err)
+	}
+
+	return pubKeyBytes, kid, nil
+}
+
+func encryptKey(wKMS KeyManager, wCrypto crypto.Crypto, cek []byte) (string, error) {
```

Comment:
You should be using `Crypto.WrapKey()` to encrypt the key.

Reply:
It seems the framework's `JWEEncrypt` does the wrapping for you... you just need to configure the remote `Crypto` implementation: https://github.com/hyperledger/aries-framework-go/blob/08fa4e3bce34a6a6798b80491f8ca4a651d83335/pkg/doc/jose/encrypter.go#L211-L215